### ISSUE TITLE
[docs]: Incorrect guide for loading NoCloud config from local filesystem.

### ISSUE THREAD

**Issue by MarekZydor** [canonical/cloud-init#5843]
# Documentation request
As described, configuration can be loaded from the local filesystem using the following format:
`ds=nocloud;s=file://path/to/directory/`

But, the prefix `file://` is inappropriate as it corrupts the path. The correct version is:
`ds=nocloud;s=/path/to/directory/`
Additionally, it is recommended to mention in the documentation that only two configuration files are required:
- user-data
- meta-data (must exist even if empty)

---
*Reported from: https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html*

**Comment by holmanb** [comment]
> the prefix file:// is inappropriate as it corrupts the path

What do you mean by "corrupts"? Are you saying that this syntax doesn't work?

**Comment by MarekZydor** [comment]
Yes, this syntax doesn't work. I spent many hours trying to figure out why the config wouldn't load. Removing that prefix fixed it.

**Comment by aciba90** [comment]
@MarekZydor, in order to assess the problem, cloud you please attach the output of `cloud-init collect-logs` from an affected instance?

**Comment by MarekZydor** [comment]
Here you go.
[cloud-init.tar.gz](https://github.com/user-attachments/files/17970285/cloud-init.tar.gz)
You can check how I passed kernel command line parameters in dmesg.txt. Also, see /run/cloud-init/status.json file, which indicates that file::// is unkown.

**Comment by aciba90** [comment]
Thanks, @MarekZydor, and apologies for the delayed response. It looks there that the seed is incorrectly specified. Your kernel cmdline is:

`KERNEL_CMDLINE=BOOT_IMAGE=/casper/vmlinuz autoinstall ds=nocloud;s=file://cdrom/nocloud/ ---`

Your path is missing a leading slash, it should be:

`file:///cdrom/nocloud/`

As is, the parsing logic is splitting after the first `/` after removing the schema part (`file://`), and considering `/nocloud` as the seed directory.
